File: code-monkey/src/tools/delete_file_tool.py
import os
from pydantic import BaseModel, Field
from typing import Type, Optional
from langchain_core.callbacks import (
    AsyncCallbackManagerForToolRun,
)
from .io_tool import IOTool
from .utils import make_file_path
from instrumentation import instrument


class DeleteFileToolInput(BaseModel):
    fname: str = Field(description="Name of the file to delete.")


class DeleteFileTool(IOTool):
    """Tool to delete a file by name"""

    name: str = "delete_file"
    description: str = "Delete a file by name"
    args_schema: Type[BaseModel] = DeleteFileToolInput

    @instrument("Tool._run", ["fname"], attributes={"tool": "DeleteFileTool"})
    def _run(
        self, fname: str, run_manager: Optional[AsyncCallbackManagerForToolRun] = None
    ) -> None:
        file_path = make_file_path(fname)

        if not os.path.exists(file_path):
            raise FileNotFoundError(f"The file {fname} does not exist.")

        os.remove(file_path)
        self.notify_file_modified(fname)
        return None

File: code-monkey-ts2/tools/delete_file_tool.ts
import * as fs from "fs";
import { z } from "zod";
import { instrument, currentSpan } from "../instrumentation";
import { IOTool } from "./io_tool";
import { CodeContext } from "../code_context";

const schema = z.object({
  fname: z.string().describe("Name of the file to delete."),
});

export class DeleteFileTool extends IOTool {
  name = "delete_file";
  description = "Delete a file by name";
  schema = schema;

  constructor(codeContext: CodeContext) {
    super(codeContext);
  }

  @instrument("Tool._call", { tool: "DeleteFileTool" })
  async _call({ fname }: z.infer<typeof schema>): Promise<string> {
    currentSpan().setAttributes({
      fname,
    });
    try {
      const filePath = this.codeContext.resolveFile(fname);
      fs.unlinkSync(filePath);
      await this.notifyFileModified(fname);
      return "file successfully deleted";
    } catch (error) {
      console.error(`Failed to delete file: ${fname}`);
      console.error(error);
      return "failed to delete file";
    }
  }
}